Initial boilerplate migration to Typescript

gomezhyuuga opened the pull request:
I'll be working with Typescript so I migrated my codebase to __`react-scripts-ts`__

Next, I'll be working on the routing.

gomezhyuuga committed:
Initial boilerplate migration to Typescript

gomezhyuuga committed:
Adding CSS Normalize

gomezhyuuga committed:
Put CSS Reset in index.css

Adding Meyer's CSS reset to the index

gomezhyuuga committed:
Adjustments of TSLint variable names rules

- Don't check too strictly variable names
- Ignore unused locals (due to react global import, it was always
warning that it wasn't used)

gomezhyuuga committed:
Initial Routing implementation, see routes.ts

Defined two routes:
- /home
- /favs
Default redirect to /home

gomezhyuuga committed:
TSLint Rule: ignore console logs warning

gomezhyuuga committed:
ADD: TSConfig downlevelIteration: true

In order to enable ES5 iteration support used by Redux Saga

gomezhyuuga committed:
Initial Redux Saga Boilerplate

Adding boilerplate files for Redux Saga implementation.

gomezhyuuga committed:
.gitignore .env file

gomezhyuuga committed:
Initial Redux markup: Wrapping app into Provider

gomezhyuuga committed:
Iniital action: FETCH_TRENDING

gomezhyuuga committed:
Giphy API functions getTrending(), search()

API#getTrending()
API#search(query: string)

gomezhyuuga committed:
GIF interface definition

Interface for a GIF Object, more details about the schema in:
https://developers.giphy.com/docs/#rendition-guide

gomezhyuuga committed:
Initial GIFList Component

Example using raw Component State

gomezhyuuga committed:
Enable Redux Chrome Extension

gomezhyuuga committed:
Extra CSS Base Styling

gomezhyuuga committed:
HomeView using Redux/Saga see reduces/index.ts

Implementation of FETCHED_TRENDING and FETCH_TRENDING

gomezhyuuga committed:
Favorite function completed

gomezhyuuga committed:
Favorite View completed

Just convert the favs object into an array and reuse GIFList. The Fav button also works out-of-the-box :)

gomezhyuuga committed:
LocalStorage implementation

Use a Middleware to store in localStorage each time the state changes.

gomezhyuuga committed:
Search functionality

Implemented Search into a Saga and into the Component

gomezhyuuga committed:
Favs Search implemented

See `reducers` and FavsSearch component

gomezhyuuga committed:
General styling improvements

gomezhyuuga committed:
Styling migration to lecture best practices

Creation of Component*.style.ts files

timgivois commented on src/components/AppNav.tsx:
```diff
@@ -0,0 +1,46 @@
+`;
+
+const LINKS = ROUTES.map(({path, content}) =>
+    <li key={path}><NavLink  activeClassName="active" to={path}>{content}</NavLink></li>);
```
try to avoid classNames

gomezhyuuga replied:
Thanks...
What could be a workaround in this case for the React Router `NavLink`?

timgivois replied:
you can create NavLink with style using styled components. https://www.styled-components.com/docs/basics#styling-any-components

gomezhyuuga replied:
Thanks!

timgivois commented on src/index.tsx:
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import App from './App';
+import './normalize.css';
```
maybe a good `injectGlobal` use case

gomezhyuuga replied:
Even for vendor content?

gomezhyuuga committed:
Initial tests for FavButton.tsx and App.tsx

bgilm commented on src/App.tsx:
```diff
@@ -0,0 +1,33 @@
+            <Header />
+            <section>
+            <Switch>
+              {ROUTES.map(({path, component, exact}) =>
```
For this specific case where we have few routes that are at the same level the benefit may not be that obvious, but remember that the mindset of React-router 4 is making routing dynamic instead of static. When defining your routes on a separate file you are still approaching this in a static way.

gomezhyuuga replied:
Alright, should I move them to this single file?
Although, the dynamic routing could still be used in any other file connecting with router right?
I'm used to use single-file definitions for routes at the same level and use multiple files (dynamic) for nested routes or with a different context.

bgilm commented on src/api.ts:
```diff
@@ -0,0 +1,22 @@
+import axios from 'axios';
+
+const API_URL = "https://api.giphy.com/v1/gifs";
+const API_KEY = process.env.REACT_APP_GIPHY_KEY;
```
👍

timgivois commented on src/styles/App.global.ts:
```diff
@@ -0,0 +1,73 @@
+	h2 { font-size: 3.6rem; line-height: 1.25; letter-spacing: -.1rem; }
+
+	/* HELPERS */
+	.center { text-align: center }
```
this could be a styled component

gomezhyuuga committed:
FavButton tests

gomezhyuuga committed:
GIFList tests

gomezhyuuga committed:
API Tests

gomezhyuuga committed:
Search tests

gomezhyuuga committed:
Fixed static typing

gomezhyuuga committed:
Improved test for GIFList

gomezhyuuga committed:
Ignore src/index.tsx due to be boilerplate

gomezhyuuga committed:
Improved Search test

gomezhyuuga committed:
FavsView.tsx tested

- Test rendering the favorites in the store
- Test rendering only those who match the filter

gomezhyuuga committed:
Redux Action Creators testing

Robfz commented on src/actions/gifs.ts:
```diff
@@ -0,0 +1,21 @@
+import { createAction } from 'redux-actions'
```
👍

Robfz commented on src/middleware/autosave.ts:
```diff
@@ -0,0 +1,12 @@
+import { Store, Action } from "redux";
+
+const autosave = (store: Store) => (next: any) => (action: Action) => {
```
👍
